src/main/kotlin/lotto/domain/Lotto.kt

```diff
@@ -1,26 +1,17 @@
 package lotto.domain
 
-class Lotto(val numbers: Set<Int> = createLottoNumbers()) {
+class Lotto(val numbers: Set<LottoNumber> = createLottoNumbers()) {
     init {
         require(numbers.size == LOTTO_NUMBER_COUNT) { "로또 당첨 번호는 6개입니다." }
-        require(checkRangeLottoNumber(numbers)) {
-            "로또 번호는 1부터 ${LOTTO_NUMBER_RANGE_END}까지 입니다."
-        }
-    }
-
-    private fun checkRangeLottoNumber(numbers: Set<Int>): Boolean {
-        return numbers
-            .filterNot { num -> num in LOTTO_NUMBER_RANGE_START until LOTTO_NUMBER_RANGE_END + 1 }
-            .isEmpty()
     }
 
     companion object {
-        const val LOTTO_NUMBER_RANGE_START = 1
-        const val LOTTO_NUMBER_RANGE_END = 45
+        private val LOTTO_NUMBER_RANGE = 1..45
         const val LOTTO_NUMBER_COUNT = 6
-        fun createLottoNumbers() =
-            (LOTTO_NUMBER_RANGE_START..LOTTO_NUMBER_RANGE_END).shuffled().subList(0, LOTTO_NUMBER_COUNT)
+        fun createLottoNumbers(): Set<LottoNumber> =
+            (LOTTO_NUMBER_RANGE).shuffled().subList(0, LOTTO_NUMBER_COUNT)
```

Comment:
```suggestion
            (LOTTO_NUMBER_RANGE).shuffled().take(LOTTO_NUMBER_COUNT)
```

`take` 함수를 사용해볼까요~?

링크
- [kotlin filter, take, drop](https://kotlinworld.com/12)